x/lockup/module.go

```diff
@@ -0,0 +1,187 @@
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	"github.com/cosmos/cosmos-sdk/x/auth/client/rest"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	"github.com/cosmos/cosmos-sdk/x/upgrade/client/cli"
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	"github.com/althea-net/althea-chain/x/lockup/keeper"
+	"github.com/althea-net/althea-chain/x/lockup/types"
+)
+
+// type check to ensure the interface is properly implemented
+var (
+	_ module.AppModule      = AppModule{}
+	_ module.AppModuleBasic = AppModuleBasic{}
+)
+
+// AppModuleBasic object for module implementation
+type AppModuleBasic struct{}
+
+// Name implements app module basic
+func (AppModuleBasic) Name() string {
+	return types.ModuleName
+}
+
+// RegisterLegacyAminoCodec implements app module basic
+func (AppModuleBasic) RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
+}
+
+// DefaultGenesis implements app module basic
+func (AppModuleBasic) DefaultGenesis(cdc codec.JSONMarshaler) json.RawMessage {
+	return cdc.MustMarshalJSON(types.DefaultGenesisState())
+}
+
+// ValidateGenesis implements app module basic
+func (AppModuleBasic) ValidateGenesis(cdc codec.JSONMarshaler, _ client.TxEncodingConfig, bz json.RawMessage) error {
+	var data types.GenesisState
+	if err := cdc.UnmarshalJSON(bz, &data); err != nil {
+		return fmt.Errorf("failed to unmarshal %s genesis state: %w", types.ModuleName, err)
+	}
+
+	return data.ValidateBasic()
+}
+
+// RegisterRESTRoutes implements app module basic
+func (AppModuleBasic) RegisterRESTRoutes(ctx client.Context, rtr *mux.Router) {
+	rest.RegisterRoutes(ctx, rtr, types.StoreKey)
+}
+
+// GetQueryCmd implements app module basic
+func (AppModuleBasic) GetQueryCmd() *cobra.Command {
+	return cli.GetQueryCmd()
+}
+
+// GetTxCmd implements app module basic
+func (AppModuleBasic) GetTxCmd() *cobra.Command {
+	return cli.GetTxCmd()
+}
+
+// RegisterGRPCGatewayRoutes registers the gRPC Gateway routes for the distribution module.
+// also implements app modeul basic
+func (AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux) {
+}
+
+// RegisterInterfaces implements app bmodule basic
+func (b AppModuleBasic) RegisterInterfaces(registry codectypes.InterfaceRegistry) {
+}
+
+//____________________________________________________________________________
+
+// AppModule object for module implementation
+type AppModule struct {
+	AppModuleBasic
+	keeper     keeper.Keeper
+	bankKeeper bankkeeper.Keeper
+}
+
+// NewAppModule creates a new AppModule Object
+func NewAppModule(k keeper.Keeper, bankKeeper bankkeeper.Keeper) AppModule {
+	return AppModule{
+		AppModuleBasic: AppModuleBasic{},
+		keeper:         k,
+		bankKeeper:     bankKeeper,
+	}
+}
+
+// Name implements app module
+func (AppModule) Name() string {
+	return types.ModuleName
+}
+
+// RegisterInvariants implements app module
+func (am AppModule) RegisterInvariants(ir sdk.InvariantRegistry) {
+	// TODO: Any invariants to check?
```

Comment:
the only invariant we might want to make sure exists is that any accounts with a funds moving exception have tokens. Obviously tehy can't get any if the lockup module is enabled.